tools/registry.py

```
@@ -29,22 +36,44 @@ def execute(self, name: str, arguments: Dict[str, Any]) -> Any:
         mapped_args = self._map_arguments(name, arguments)
 
+        start_time = time.time()
+        error = None
+        result = None
+
         try:
-            return tool["function"](**mapped_args)
+            result = tool["function"](**mapped_args)
         except ToolError:
             raise
         except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)